### ISSUE TITLE
Periodic system/process metricset crashes under Windows

### ISSUE THREAD

**Issue by adriansr** [elastic/beats#12826]
- Version: 7.2.0
- Operating System: Windows
- Discuss Forum URL: -
- Steps to Reproduce: 

Users report a periodic crash in Metricbeat under Windows, when the `system/process` metricset has been running for some hours. After enabling [logging.files.redirect_stderr](https://www.elastic.co/guide/en/beats/metricbeat/current/configuration-logging.html#_literal_logging_files_redirect_stderr_literal_phrase_role_experimental_phrase) a crash similar to this appears in the logs:

```
signal arrived during external code execution

syscall.Syscall(0x7ffb7d55ce00, 0x2, 0xc000ffff80, 0xc000e30f74, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/runtime/syscall_windows.go:184 +0xea
syscall.CommandLineToArgv(0xc000ffff80, 0xc000e30f74, 0x80, 0x0, 0x0)
	/usr/local/go/src/syscall/zsyscall_windows.go:930 +0x80
github.com/elastic/beats/vendor/github.com/elastic/gosigar/sys/windows.ByteSliceToStringSlice(0xc000ffff80, 0x80, 0x80, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/github.com/elastic/beats/vendor/github.com/elastic/gosigar/sys/windows/syscall_windows.go:525 +0x75
github.com/elastic/beats/vendor/github.com/elastic/gosigar.(*ProcArgs).Get(0xc000e31240, 0x170, 0x0, 0x0)
	/go/src/github.com/elastic/beats/vendor/github.com/elastic/gosigar/sigar_windows.go:376 +0x1f1
github.com/elastic/beats/libbeat/metric/system/process.(*Process).getDetails(0xc000a9b040, 0xc000e313b0, 0xb, 0xc0008a5401)
	/go/src/github.com/elastic/beats/libbeat/metric/system/process/process.go:138 +0x63c
github.com/elastic/beats/libbeat/metric/system/process.(*Stats).getSingleProcess(0xc000964510, 0x170, 0xc00025de60, 0x8)
	/go/src/github.com/elastic/beats/libbeat/metric/system/process/process.go:484 +0x240
github.com/elastic/beats/libbeat/metric/system/process.(*Stats).Get(0xc000964510, 0xc000c22390, 0x0, 0xc0007d7710, 0x0, 0x0)
	/go/src/github.com/elastic/beats/libbeat/metric/system/process/process.go:425 +0x113
github.com/elastic/beats/metricbeat/module/system/process.(*MetricSet).Fetch(0xc000980000, 0x9fd0a80, 0xc0008a5170)
	/go/src/github.com/elastic/beats/metricbeat/module/system/process/process.go:102 +0x51
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).fetch(0xc000954e60, 0x254d1a0, 0xc0008a5170)
	/go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:238 +0x2b7
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).startPeriodicFetching(0xc000954e60, 0x254d1a0, 0xc0008a5170)
	/go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:219 +0x121
github.com/elastic/beats/metricbeat/mb/module.(*metricSetWrapper).run(0xc000954e60, 0xc000a1c060, 0xc000a06180)
	/go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:196 +0x676
github.com/elastic/beats/metricbeat/mb/module.(*Wrapper).Start.func1(0xc000a04050, 0xc000a1c060, 0xc000a06180, 0xc000954e60)
	/go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:137 +0x27e
created by github.com/elastic/beats/metricbeat/mb/module.(*Wrapper).Start
	/go/src/github.com/elastic/beats/metricbeat/mb/module/wrapper.go:125 +0x147
```

**Comment by cpiment** [comment]
@adriansr thanks for the patch for 7.2. Is there any way we can download the windows binaries with this patch merged? Or the only way is to build the binaries?

**Comment by adriansr** [comment]
@cpiment until 7.2.1 is out (next Tuesday?), the only way will be to provide them with patched binaries. Are they experiencing crashes in beats other than Metricbeat?

**Comment by cpiment** [comment]
We are using heartbeat, winlogbeat and metricbeat (all 7.2.0), but metricbeat is the only one that crashes.

**Comment by adriansr** [comment]
@detro you'll have to wait until 7.2.1 is released or build your own 7.2.1-snapshot (is not complicated at all). Are you having this same issue? It's a Windows-only crash and has only affected Metricbeat so far. Please share if you're experiencing otherwise.

**Comment by detro** [comment]
Sorry for the mistake: I meant to comment on this one https://github.com/elastic/beats/pull/12268 - I think in digging into the git history and trying to figure out when the issue was fixed, I took a wrong turn and then commented in the wrong place. This is really about skipping malformed docker log lines. Apologies for the confusion.

In regards to building a snapshot, is there a pointer to a Dockerfile I could use for this?
